src/components/date-picker/date-picker-dropdown.tsx

```diff
@@ -0,0 +1,68 @@
+import * as React from 'react';
+import {getMonthNames, getMonthFromOffset} from './date-picker-helpers';
+import {observable, action, computed} from 'mobx';
+import {observer} from 'mobx-react';
+import {DatePickerGrid} from './date-picker-grid';
+import styles from './date-picker.st.css';
+
+export interface DatePickerDropdownProps {
+    date: Date;
+    onChange (date: Date): void;
+}
+
+const monthNames = getMonthNames();
+
+@observer
```

Comment:
the date picker grid is actually a part which will be better optimized with mobx.